### PR TITLE
Muelu: cleanup coarse map factories

### PR DESCRIPTION
**Pull request by mayrmt**
@trilinos/muelu

## Motivation

While revisiting MueLu in the context of blocked systems, I made some small cleanups in `CoarseMapFactory` and `BlockedCoarseMapFactory`.

Changes include:
- adding / fixing documentation
- adding more information to some error messages
- adding `Constructor` unit tests
- removing lots of training white spaces

I also reverted a recent change (PR #7777) to the ordering of `typedef` in `MueLu_UseShortNamesScalar.hpp` to be consistent with the auto-generated ordering.

## Related Issues

* Related to #7777

## Testing

`ctest` passes locally. Let the autotester check for the rest.

## Additional Information

These changes are part of a larger effort to MueLu for blocked systems arising in surface-coupled problems. This PR separates these rather general changes from the more specific ones yet to come in the near future. This split into multiple PRs aims at making the change set smaller and the code review easier.

### CODE REVIEW

File: packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_CoarseMapFactory_def.hpp [hunk 1]
```diff
@@ -126,7 +117,8 @@ namespace MueLu {
     RCP<const Teuchos::Comm<int> > comm    = aggregates->GetMap()->getComm();
 
     // read in offset information from parameter list and fill the internal member variable
-    GlobalOrdinal domainGidOffset = 0;
+    const GlobalOrdinal GO_ZREO = Teuchos::ScalarTraits<GlobalOrdinal>::zero();
```
Comment (jhux2) [inline packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_CoarseMapFactory_def.hpp:120]: This should be "GO_ZERO".

**Review by jhux2 — CHANGES_REQUESTED** [review]
@mayrmt Looks good to me, but just a couple change requests.

File: packages/muelu/src/Headers/MueLu_UseShortNamesScalar.hpp [hunk 2]
```diff
@@ -146,15 +146,15 @@ typedef MueLu::HierarchyFactory<Scalar,LocalOrdinal,GlobalOrdinal,Node> Hierarch
 #ifdef MUELU_HIERARCHYUTILS_SHORT
 typedef MueLu::HierarchyUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node> HierarchyUtils;
 #endif
+#ifdef MUELU_INTERFACEAGGREGATIONFACTORY_SHORT
+typedef MueLu::InterfaceAggregationFactory<Scalar,LocalOrdinal,GlobalOrdinal,Node> InterfaceAggregationFactory;
+#endif
```
[the remark this answers is not in the document]
Comment (mayrmt) [reply packages/muelu/src/Headers/MueLu_UseShortNamesScalar.hpp:151]: That is my fault. I accidentally changed the ordering in #7777. The current ordering is now generated by the `packages/muelu/src/Utils/ClassList/gen_all.sh` script, so it should be reproducible.

**Review by jhux2 — DISMISSED** [review]
lgtm

**Comment by mayrmt** [comment]
@jhux2 I'll have to enable some more tests in `BlockedCoarseMapFactory.cpp`, but this will probably require some more rounds in the auto-tester. Then, I'll squash my commits, before merging.

**Pull request closed without merging**